database/Entities/User.php

```diff
@@ -912,6 +915,32 @@ public function is2faEnabled()
         return $this->getUser2FA() && $this->getUser2FA()->enabled();
     }
 
+    /**
+     * Check if the user is 2FA authenticated
+     *
+     * * return true if the user has a 2FA create but not enabled because
+     * * the GoogleAuthenticator->isAuthenticated => canPassWithoutCheckingOTP() => isActivated() just check that the 'secret' filed in DB is set and
+     * * does not check if the filed 'enabled' is true or false
+     *
+     * * if the user has a 2FA create and enabled then we check if GoogleAuthenticator is authenticated
+     *
+     * @return bool
+     */
+    public function is2faAuthenticated()
```

Comment:
@yannrobin - this function name is confusing. I we go with this we'll be scratching out heads in a year's time.

Maybe: `is2faAuthRequiredForSession()` - as we really asking whether the user needs to authenicate via 2fa for this session.

Then the function result needs to be reversed.